src/test/functional/functional-tests.js

```diff
@@ -1940,53 +1902,115 @@ describe('functional tests', function() {
+            })
+            response.on('data', () => {
+              // just drain
+            })
+          })
+
+          uploadRequest.on('error', (er) => {
+            done(er)
+          })
+
+          uploadRequest.write(_100_byte)
+          uploadRequest.end()
         })
       }else {
         done()
       }
+
     })
 
-    step(`presignedPutObject(bucketName, objectName, expires=1000, _incoming_obj,cb)_bucketName:${versionedBucketName} ${versionedPresignObjName} _version 2`, done => {
+    step(`presignedPutObject(bucketName, objectName, expires=1000, cb)_bucketName:${versionedBucketName} ${versionedPresignObjName} _version:2`, done => {
       if(isVersioningSupported) {
-        putPreSignedObject(versionedBucketName,versionedPresignObjName,1000,_200_byte,()=>{
-          done()
+        client.presignedPutObject(versionedBucketName,versionedPresignObjName,1000, (e, presignedUrl) => {
+          if (e) {
+            done(e)
+          }
+          let mobileClientReqWithProtocol = http
+          var upldRequestOptions = _.pick(url.parse(presignedUrl), ['hostname', 'port', 'path', 'protocol'])
+          upldRequestOptions.method = 'PUT'
+          upldRequestOptions.headers = {
+            'content-length': _200_byte.length
+          }
+          if (upldRequestOptions.protocol === 'https:') {
+            mobileClientReqWithProtocol = https
+          }
+          const uploadRequest = mobileClientReqWithProtocol.request(upldRequestOptions, (response) => {
+            if (response.statusCode !== 200) return new Error(`error on put : ${response.statusCode}`)
+            response.on('error', (err) => {
+              done(err)
+            })
+            response.on('end', () => {
+              done()
+            })
+            response.on('data', () => {
+              // just drain
+            })
+          })
+
+          uploadRequest.on('error', (er) => {
+            done(er)
+          })
+
+          uploadRequest.write(_200_byte)
+          uploadRequest.end()
         })
       }else {
         done()
       }
     })
 
-    step(`listObjects(bucketName, objectName, expires=1000, _incoming_obj,cb)_bucketName:${versionedBucketName} ${versionedPresignObjName} _version 2`, done => {
+    step(`listObjects(bucketName, '', true, {IncludeVersion: true}, cb)_bucketName:${versionedBucketName}  _prefix:""`, done => {
       if(isVersioningSupported) {
         const objectsStream = client.listObjects(versionedBucketName, '', true,{IncludeVersion: true})
         objectsStream.on('data', function(obj) {
-          objectsList.push({VersionId:obj.versionId, Key:obj.name})
+          objectsList.push({versionId:obj.versionId, name:obj.name})
         })
 
         objectsStream.on('error', function() {
           return done()
         })
         objectsStream.on('end', function() {
-          if(objectsList.length === 2) {
+          const versionCount = objectsList.length
+          if(versionCount === 2) {
```

Comment:
It may be a good idea to create a constant for the expected version count, instead of hard coding it,`2`. The same constant can also be used to assert when removing the objects as in the old implementation.

You can ignore this comment if it sounds superfluous to you, but I'd prefer `noOfVersionedObjects instead of `versionCount`.

Reply:
Thank you so much. I have updated.